qrcode_detection_package/main.py

```diff
@@ -1,56 +1,358 @@
+            QRCodeInfo, 'qr_code_info', 10)
+
+        # until tested properly different detection methods are implemented and can be configured here
+        # 0 = load stored test image (for testing purposes)
+        # 1 = use OpenCV to get image from camera (this is the wanted solution)
+        # 2 = use libcamera shell script to take photo and load it (fallback solution)
+        self.config_detection_method = 0
+
+    def __callback_control(self, control_msg):
+        """
+        Callback function for receiving control messages.
+
+        This callback function is triggered when a control message is received.
+        It checks if the target ID of the control message matches the QR code scanner node.
+        If it does, the function logs a message indicating the receipt of the control message.
+        Depending on the 'active' flag in the control message, the function either activates
+        or deactivates the QR code scanner node by calling the respective private methods.
+
+        Args: 
+            control_msg: The received control message.
+
+        Returns: 
+            None
+        """
+        if (control_msg.target_id == "qr_code_scanner_node"):
+            self.get_logger().info("Recieved control message")
+            if (control_msg.active):
+                self._activate_()
+            else:
+                self._deactivate_()
+
+    def set_state(self, state):
+        """
+        Sets the state of the node to the new state provided.
+
+        Args: 
+            state: The new state to set the node to.
+
+        Returns: 
+            None
+        """
+        self.node_state = state
 
     def __capture_image(self):
-        # For testing purposes the image is not taken from the camera but loaded a test file
-        # Path to the image file
-        image_path = 'src/qrcode_detection_package/test_image/test.jpg'
-        # Load the image using OpenCV
-        captured_image = cv2.imread(image_path)
+        """
+        Capture an image either from the camera or a test image, depending on the configuration.
+
+        This method captures an image either from the camera or from a test image,
+        depending on the configuration specified by 'config_detection_method'.
+        - If 'config_detection_method' is set to 0, a test image is loaded from a predefined path.
+        - If 'config_detection_method' is set to 1, an image is captured from the camera.
+        - If 'config_detection_method' is set to 2, an image is captured using the libcamera command line tool.
+        The captured image is returned as an OpenCV MatLike object.
+
+        Returns: 
+            OpenCV MatLike representing the captured image.
+        """
+        import cv2
+
+        # Check which detection style should be used
+        if (self.config_detection_method == 0):
+            # Path to the test image
+            test_image_path = 'src/qrcode_detection_package/test_image/test3.png'
```

Comment:
You could maybe add a function here tgat returns the path to a random working image as to not rely to heavily on a single image. This can be done later tho or left out entirely if done via unit tests instead.

Reply:
yes, i am planning such a randomiser for the used image for unit testing, as that option is for testing purposes only anyways.